src/Business Foundation/App/NoSeries/src/Batch/NoSeriesBatchImpl.Codeunit.al

```diff
@@ -17,6 +17,11 @@ codeunit 309 "No. Series - Batch Impl."
         SimulationMode: Boolean;
         CannotSaveNonExistingNoSeriesErr: Label 'Cannot save No. Series Line that does not exist: %1, %2', Comment = '%1 = No. Series Code, %2 = Line No.';
         CannotSaveWhileSimulatingNumbersErr: Label 'No. Series state cannot be saved while simulating numbers.';
+        NoSeriesBatchLbl: Label 'No. Series - Batch', Locked = true;
+        SimulationModeStartedTxt: Label 'No. Series simulation mode started.', Locked = true;
+        SavingSingleNoSeriesStateTxt: Label 'Saving single No. Series state.', Locked = true;
+        SavingAllNoSeriesStatesTxt: Label 'Saving all No. Series states.', Locked = true;
+        UpdatingNoSeriesLinesFromDbTxt: Label 'Updating No. Series lines from database.', Locked = true;
```

Comment:
Since all labels are locked, what about renaming them as `*Tok`?
(Applies to all other new ones in this PR as well.)

Reply:
👍 Tok is meant to be used when the value of the label matches the name. As in: TokenTok: Label 'Token', Locked = true;
For these ones I guess we're close enough :-)

Reply:
@grobyns 
Can you please explain the `Tok` rule? As I understood, the Label name should always reflect its content, at least more or less. Furthermore, the [example in the docs](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0074) does the exact opposite of what you've just stated.
The community understood Tokens to be used this way: https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0047 - wrong?

Reply:
Just discussed with Nikola and Tok are for simple variables where the entire string is also part of the label, which I guess would work for the NoSeriesBatchTok. We are also looking into if we could add some other extension for these locked labels. I will go for Txt then, since 'No. Series simulation mode started.' is not really a token nor a label, it's a description of what is happening. I.E. a Text. Once we get this locked extension we can switch to that.

Reply:
@NKarolak the docs looks to be out of date, I believe the example should be: GetTok: Label 'GET', Locked = true;
Will create a bug for that.

Reply:
@AndreasMoth Thanks! While you are at it, could you also add some more explanations on when to use what, typically?

Reply:
@AndreasMoth, bit of a side question here, but may I ask what would be your advice for the community on having a Code Analyzer for applying the right suffix on handeling labels?

As @NKarolak mentioned we've created two rules in our Community Code Analyzer; BusinessCentral.LinterCop

* [LC0046 - Label with suffix Tok must be locked.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0046)
* [LC0047 - Locked Label must have a suffix Tok.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0047)

With the information here I guess the LC0047 rule is invalid? Is it a good idea to adapt this to "the entire string is also part of the label", like the GetTok: Label 'GET' example?

Reply:
@NKarolak @Arthurvdv I'm not going to try dictate any policies in a PR, but the way I go about it is, there are first some easy ones:
Err for Error messages "Error(NoLinesErr)"
Msg for Messages "Message(InvoicePostedMsg)"
Qst for questions in confirm dialogs "if Confirm(PostInvoiceQst) then"
Tok for any short tokens such as "GET", "PUT", "HTTPS" etc. where the name would then also align with the label itself, now these would generally always also be locked unless there's a very good reason.
Lbl and Txt I don't have a good explanation for.

I think as for these LC rules, I guess it depends on how partners wants to work with this. I can see these rules can help partners align their code better even if we don't have these.

I hope this answers all questions :). I do think however it's important to use what works for you as a partner.